google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java

```diff
@@ -266,7 +266,18 @@ static Object convertToPlainJavaTypes(Object object) {
       Type genericType = type.getActualTypeArguments()[0];
       if (o instanceof List) {
         List<Object> list = (List<Object>) o;
-        List<Object> result = new ArrayList<>(list.size());
+        List<Object> result = null;
+        try {
+          result =
+              (rawType == List.class)
+                  ? new ArrayList<>(list.size())
+                  : (List<Object>) rawType.getDeclaredConstructor().newInstance();
+        } catch (InstantiationException
+            | IllegalAccessException
+            | NoSuchMethodException
+            | InvocationTargetException e) {
+          throw new RuntimeException(e);
```

Comment:
Just FYI: While uncommon, I think this method would now throw if called on an type that represents an subinterface of List, e.g.:

interace MyList extends List {
}

class Pojo {
  MyList data;
}
```

I am however not 100% certain that is the case, but based on my reading of https://stackoverflow.com/questions/36564041/getdeclaredconstructor-on-an-interface, this would throw an NoSuchMethodException, whereas before it would just create an ArrayList (albeit of the wrong type). I don't know if there is a solution that doesn't involve creating a list of the wrong type, and if there isn't, we can probably leave this logic as is.